cmd/gateway/b2/gateway-b2.go

```diff
@@ -460,21 +459,32 @@ func (l *b2Objects) GetObjectInfo(ctx context.Context, bucket string, object str
 	if err != nil {
 		return objInfo, err
 	}
-	f, err := bkt.DownloadFileByName(l.ctx, object, 0, 1)
+
+	f, _, err := bkt.ListFileNames(l.ctx, 1, object, "", "")
 	if err != nil {
 		logger.LogIf(ctx, err)
 		return objInfo, b2ToObjectError(err, bucket, object)
 	}
-	f.Close()
-	fi, err := bkt.File(f.ID, object).GetFileInfo(l.ctx)
+
+	if len(f) != 1 {
+		err := minio.ObjectNotFound{
+			Bucket: bucket,
+			Object: object,
+		}
+
+		logger.LogIf(ctx, err)
```

Comment:
I don't think we need to log this error, since it is totally normal to not find an object if the user enters a wrong file name.

Reply:
👍 I'll sort these out later today, I've also noticed an issue with deletion not behaving as expected due to B2's versioning which I'll look to address.